packages/eddsa-proof/src/generate.ts

```diff
@@ -30,7 +32,7 @@ export default async function generate(
 
     // allow user to override our artifacts
     // otherwise they'll be downloaded if not already in local tmp folder
-    snarkArtifacts ??= await maybeGetEdDSASnarkArtifacts()
+    snarkArtifacts ??= await maybeGetSnarkArtifacts("eddsa")
```

Comment:
> Do you mean that there would be artifacts added to https://github.com/privacy-scaling-explorations/snark-artifacts that don't (and maybe will never) have a dedicated package in https://github.com/privacy-scaling-explorations/zk-kit (like there would be a @zk-kit/someProof-artifacts package but no @zk-kit/someProof-proof package).

Just discussed this privately. Yes, that's what I have in mind.

> I still like the idea of avoiding free text and restrict the param to valid values: those artifacts we are able to provide artifacts for.

We agreed on avoiding free text and creating an enum for each new snark-artifacts package.

> In any case we would have some work/PR to review in https://github.com/privacy-scaling-explorations/snark-artifacts no: for the PR that adds the artifact. So maybe the enum belongs to the snark-artifacts repo and any PR that adds artifacts would need to update the enum too?

We thought of having a new `@zk-kit/artifacts` package -> https://github.com/privacy-scaling-explorations/snark-artifacts/issues/11#issuecomment-2088149693. Where should it stay, zk-kit or snark-artifacts repo?

Reply:
I would put it in the snark-artifacts repo. Just because I kinda "feel" that enum is more closely related to the artifacts themselves (the content of the snark artifacts repo) rather than to all the other zk-kit packages.

Reply:
Makes sense! I'll wrap this PR up, and then we can open a new one to move the code there.